x/escrow/handler.go

```diff
@@ -270,13 +277,22 @@ func (h ReturnEscrowHandler) Deliver(ctx weave.Context, db weave.KVStore,
 		return res, err
 	}
 
-	// move the money from escrow to recipient
-	dest := weave.Address(escrow.Sender)
-	if err := h.ops.Withdraw(db, escrow, key, dest, escrow.Amount); err != nil {
+	// query current escrow balance
+	escrowAddr := Condition(key).Address()
+	available, err := h.bank.Balance(db, escrowAddr)
+	if err != nil {
 		return res, err
 	}
-	// returns error if Delete failed
-	return res, err
+
+	// withdraw all coins from escrow to the defined "sender"
+	dest := weave.Address(escrow.Sender)
+	for _, c := range available {
```

Comment:
again, move coins function please

Reply:
👍 I noticed this duplication, too. Then thought we may upgrade the coin mover interface for this but reverted to not blow up this PR. WDYT? I will start with the function in the handler